isis/tests/Statistics.cpp

```diff
@@ -0,0 +1,143 @@
+#include "Statistics.h" 
+
+#include "FileName.h"
+#include "IException.h"
+#include "Preference.h"
+#include "Statistics.h"
+#include "XmlStackedHandlerReader.h"
+
+#include <QDebug>
+#include <QDataStream>
+#include <QXmlStreamWriter>
+#include <QXmlInputSource>
+
+#include <float.h>
+
+#include <gtest/gtest.h>
+
+using namespace Isis;
+using namespace std;
+
+#define TEST_EPSILON 1e-8
+
+/**
+ * @author 2018-11-20 Tyler Wilson
+ *
+ * @internal
+ *   @history 2018-11-20 Tyler Wilson - Coverage of existing
+ *            statistical routines using truth data generated
+ *            in R.
+ */
+
+
+
+TEST(Statistics, HandCalculations) {
+
+    Statistics s;
+    
+    s.AddData(-10.0);
+    s.AddData(20.0);
+    s.AddData(30.0);
+    
+
+    //Values were computed in R
+    double mu =13.3333333333333;
+    double std = 20.8166599946613;
+    double var = 433.333333333333;
+    
+    double rms = 21.6024689946929;   
+
+    EXPECT_NEAR(s.Average(),mu,TEST_EPSILON);
+    EXPECT_NEAR(s.Variance(),var,TEST_EPSILON);
+    EXPECT_NEAR(s.StandardDeviation(),std,TEST_EPSILON);
+    EXPECT_NEAR(s.Sum(),40.0,TEST_EPSILON);
+    EXPECT_NEAR(s.SumSquare(),1400.0,TEST_EPSILON);
+    EXPECT_NEAR(s.ValidPixels(),3.0,TEST_EPSILON);
+    EXPECT_NEAR(s.Rms(),rms,TEST_EPSILON);    
+    EXPECT_NEAR(s.Minimum(),-10.0,TEST_EPSILON);
+    EXPECT_NEAR(s.Maximum(),30.0,TEST_EPSILON);
+    EXPECT_NEAR(s.ZScore(1.0),(1-mu)/std,TEST_EPSILON);    
+    double percent = 99.5;    
+    double k = sqrt(1.0 / (1.0 - percent / 100.0));
+    EXPECT_NEAR(s.ChebyshevMinimum(),mu -k*std,TEST_EPSILON);
+    EXPECT_NEAR(s.ChebyshevMaximum(),mu +k*std,TEST_EPSILON);
+
+   
+}
+
+TEST(Statistics,SpecialPixels) {
+
+  Isis::Statistics t;
+
+    double a[10];
+    a[0] = 1.0;
+    a[1] = 2.0;
+    a[2] = 3.0;
+    a[3] = NULL8;
+    a[4] = HIGH_REPR_SAT8;
+    a[5] = LOW_REPR_SAT8;
+    a[6] = HIGH_INSTR_SAT8;
+    a[7] = LOW_INSTR_SAT8;
```

Comment:
Is there a reason why you're using the ___8 names instead of just Null, Hrs, ... ?

Reply:
No reason.  I changed them to Hrs,Lrs,...etc.